fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/typing/Type.java

```diff
@@ -434,6 +434,8 @@ private static Type fromProtoType(@Nullable Descriptors.GenericDescriptor descri
                 final var elementField = messageDescriptor.findFieldByName(NullableArrayTypeUtils.getRepeatedFieldName());
                 final var elementTypeCode = TypeCode.fromProtobufType(elementField.getType());
                 return fromProtoTypeToArray(descriptor, protoType, elementTypeCode, true);
+            } else if (TupleFieldsProto.UUID.getDescriptor().equals(messageDescriptor)) {
```

Comment:
I think we may have discussed this before, but can you please remind me again why are handling the conversion of `UUID` PB message within a `Struct` but not within and `Array`?

Reply:
We are adding an extra check here for the case of `RECORD`, since the descriptor could be that representing a `UUID`. If so, we do not consider it a "Record" type, but a "UUID" type. This is only needed for this particular case, and not also in Arrays as that dispatches the element processing to this function only.